src/main/java/com/kodilla/ecommercee/controller/CartController.java

```diff
@@ -18,23 +22,44 @@
 public class CartController {
 
     @Autowired
-    private ProductRepository productRepository;
+    private CartService cartService;
+
+    @Autowired
+    private CartMapper cartMapper;
+
+    @Autowired
+    private OrderMapper orderMapper;
+
+    @Autowired
+    private OrderService orderService;
 
     @PostMapping(value = "createNewCart", consumes = APPLICATION_JSON_VALUE)
     public void createNewCart(@RequestBody CartDto cartDto) {
+        cartService.addCart(cartMapper.mapToCart(cartDto));
     }
+
     @GetMapping(value = "getProductsFromCart")
-    public List<Product> getProductsFromCart(@RequestParam("cartId") Long cartId) {
-        return new ArrayList<> ();
+    public CartDto getProductsFromCart(@RequestParam("cartId") Long cartId) {
+        return cartMapper.mapToCartDto(cartService.getCart(cartId).orElseThrow(CartNotFoundException::new));
     }
+
     @PostMapping(value = "addProductToCart", consumes = APPLICATION_JSON_VALUE)
     public void addProductToCart(@RequestParam ("cartId") Long cartId, @RequestParam("productId") Long productId) {
+        Cart cart = cartService.getCart(cartId).orElseThrow(CartNotFoundException::new);
+        Product product = cartService.findProduct(productId).orElseThrow(CartNotFoundException::new);
+        cart.getProducts().add(product);
     }
+
     @DeleteMapping(value = "deleteProductFromCart")
-    public void deleteProductFromCard(@RequestParam ("cartId") Long cartId, @RequestParam("productId") Long productId) { }
+    public void deleteProductFromCard(@RequestParam ("cartId") Long cartId, @RequestParam("productId") Long productId) {
+        Cart cart = cartService.getCart(cartId).orElseThrow(CartNotFoundException::new);
+        Product product = cartService.findProduct(productId).orElseThrow(CartNotFoundException::new);
+        cart.getProducts().remove(product);
+    }
     
     @PostMapping(value = "createOrder", consumes = APPLICATION_JSON_VALUE)
-    public void createOrder(@RequestParam ("cartId") Long cartId) {
+    public OrderDto createOrder(@RequestParam ("order") Order order) {
+        return orderMapper.mapToOrderDto(orderService.saveOrder(order));
     }
```

Comment:
Teraz tworzysz po prostu nowe zamówienie niezależne od koszyka. Sugeruję wykonać to w taki sposób, że lista produktów z koszyka, trafi do listy produktów z zamówienia.